topo/node/cisco/cisco.go

```diff
@@ -148,18 +157,112 @@ func (n *Node) Create(ctx context.Context) error {
 		return fmt.Errorf("failed to create pod for %q: %w", pb.Name, err)
 	}
 	log.Debugf("Pod created:\n%+v\n", sPod)
-	log.Infof("Created XRD node resource %s pod", n.Name())
+	log.Infof("Created Cisco %s node resource %s pod", n.Proto.Model, n.Name())
 	if err := n.CreateService(ctx); err != nil {
 		return err
 	}
-	log.Infof("Created XRD node resource %s services", n.Name())
+	log.Infof("Created Cisco %s node resource %s services", n.Proto.Model, n.Name())
 	return nil
 }
 
+func constraints(pb *tpb.Node) map[string]string {
```

Comment:
why not properly handle defaults in this function rather then having the if stanza's below?

	constraints := constraints(pb)
	if pb.Constraints == nil {
		pb.Constraints = constraints
	} else {
		if pb.Constraints["cpu"] == "" {
			pb.Constraints["cpu"] = constraints["cpu"]
		}
		if pb.Constraints["memory"] == "" {
			pb.Constraints["memory"] = constraints["memory"]
		}
	}

Reply:
done